### ISSUE TITLE
Contact Journal (Diary) CSV import

### ISSUE THREAD

**Issue by heeplr** [corona-warn-app/cwa-wishlist#324]
## Current Implementation
Thank you for implementing the Contact Journal. CSV export already works.

## Suggested Enhancement
A feature to import contacts from a CSV file would be useful.

## Expected Benefits
A lot of people used other contact diary apps to keep track of their contacts before the feature was introduced in CWA (and might want to continue using them).
Those apps support the export to a CSV file but don't offer features, the CWA offers.
When there is an infection, one might want to use the contact journal of the CWA but would be forced to create all entries manually. A CSV import would greatly ease that process, even if the format of the CSV doesn't match.

It would be easier to convert the CSV manually (e.g. in Excel) and then import it into the CWA, than transferring every contact manually one-by-one.

**Comment by jucktnich** [comment]
14 days will be gone when this feature is implemented, so it's not needed anymore.

**Comment by heeplr** [comment]
As mentioned, people might continue to use other apps and only want to import into CWA in case of infection. In my case that's true especially for elderly users who are not agile enough to switch their newly learned habits. They just won't transfer contacts into CWA. I'd see it as a great improvement if a tech savy relative can import contacts into CWA as easy as possible.

Also other Apps might support features that CWA wouldn't need to implement.

@jucktnich hence I disagree that this feature isn't needed, because of the 14 day timespan.
If such a feature would impose a major workload, I'd agree to not make it top priority, of course.

**Comment by Ein-Tim** [comment]
@heeplr 

Are the CSV files from every contact diary structured the same way?
Because otherwise this gets a bit hard to implement

**Comment by heeplr** [comment]
@Ein-Tim They all contain contacts and places.
I'm sure they are not structured the same way, but if the CWA could import it's own structure (the one it exports), one could relatively simply convert it (e.g. in Excel). That is a feature of the CSV format.

Also, since CWA is a kind of lighthouse among contact tracing apps, I could imagine that other Contact Journal Apps will add support for exporting the CWA structure. 

Of course there would be no motivation for 3rd party developers to implement that, if  CWA doesn't support import in the first place.

**Comment by Ein-Tim** [comment]
Okay thank you. 
I think any reformatting from user side should not be part of the process (or only a first solution).

The developers from CWA could simply ask the user which contact diary app they are using and then CWA knows where to look for the right data. But this is a very fragile solution since 3rd party developers could always change the CSV file from version to version... 

So or so let's wait for the community managers picking up on this, related Issues are #311 & #306. 

Have a nice evening and stay safe!

**Comment by heeplr** [comment]
@Ein-Tim You are right that this is only a compromise but supporting every single CSV format out there would be too much to ask for in my opinion (CWA was never meant to do that, anyway).

Maybe we could see it that way: Just support import of the own CWA format since I can export contacts already when I change phones or uninstall the app. But I can't import it on a new phone when re-installing CWA. If that is supported, the feature would be complete.

Then, as an **additional** bonus:
* users could import 3rd party data by converting data manually
* 3rd party developers can pick up the feature and support exporting CWA format (as long as CWA doesn't change the structure very often, which should be avoided if possible).

Thank you all so much for your work & stay safe as well.

**Comment by jucktnich** [comment]
@heeplr I meant if you start now with using the cwa contact diary, in 14 days you can delete the other apps, because you're contacts get deleted anyway, and I think, there's no way, to get this feature before.
If the diary isn't safed to the backup, I would appreciate this feature.

**Comment by dsarkar** [comment]
Internal Tracking ID: EXPOSUREAPP-3034
